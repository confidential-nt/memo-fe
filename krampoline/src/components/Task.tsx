import React from "react";
import Checkbox from "@mui/material/Checkbox";
import { PiTrashLight, PiPencilLight } from "react-icons/pi";

function Task({ todo, onUpdate, onDelete }) {
  const { id, title, status } = todo;

  const handleChange = (e) => {
    onUpdate({ ...todo, status: e.target.checked ? "completed" : "active" });
  };

  const handleDelete = () => onDelete(todo);

  return (
    <li className="flex justify-between items-center w-30 h-12 shrink-0 border-black border-2 rounded-[15px] shadow-standard bg-white my-4">
      <div className="flex justify-left items-center">
        <Checkbox
          type="checkbox"
          id={id}
          checked={status === "completed"}
          onChange={handleChange}
          defaultChecked
          color="secondary"
          className="cursor-pointer"
        />
        <label htmlFor={id} className="flex justify-left items-center">
          {title}
        </label>
      </div>
      <span className="flex justify-right items-center">
        <PiPencilLight className="w-6 h-6 shrink-0 mx-1 cursor-pointer" />
        <PiTrashLight
          onClick={handleDelete}
          className="w-6 h-6 shrink-0 mx-1 cursor-pointer"
        />
      </span>
    </li>
  );
}

export default Task;
