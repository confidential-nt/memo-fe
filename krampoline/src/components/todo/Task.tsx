import { ChangeEvent } from "react";
import Checkbox from "@mui/material/Checkbox";
import { PiTrashLight, PiPencilLight } from "react-icons/pi";
import { TodoItem } from "../../pages/Todo";

type Props = {
  todo: TodoItem;
  onUpdate: (todo: TodoItem) => void;
  onDelete: (todo: TodoItem) => void;
};

function Task({ todo, onUpdate, onDelete }: Props) {
  const { id, content, status } = todo;

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    onUpdate({ ...todo, status: e.target.checked ? "completed" : "active" });
  };

  const handleDelete = () => onDelete(todo);

  return (
    <li className="flex justify-between items-center w-30 h-12 shrink-0 border-black border-2 rounded-[15px] shadow-standard bg-white my-4">
      <div className="flex justify-left items-center">
        <Checkbox
          id={id}
          checked={status === "completed"}
          onChange={handleChange}
          color="secondary"
          className="cursor-pointer"
        />
        <label
          htmlFor={id}
          className="flex justify-left items-center"
          style={{
            textDecoration: status === "completed" ? "line-through" : "none",
          }}
        >
          {content}
        </label>
      </div>
      <span className="flex justify-right items-center">
        <PiPencilLight className="w-6 h-6 shrink-0 mx-1 cursor-pointer" />
        <PiTrashLight
          onClick={handleDelete}
          className="w-6 h-6 shrink-0 mx-1 cursor-pointer"
        />
      </span>
    </li>
  );
}

export default Task;
