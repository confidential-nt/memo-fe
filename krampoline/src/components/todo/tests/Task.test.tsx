import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";
import Task from "../Task";

describe("Task Test", () => {
  it("스냅샷 테스트", () => {
    const { asFragment } = renderTask();
    expect(asFragment()).toMatchSnapshot();
  });

  it("할 일이 표시되어야한다", () => {
    renderTask();
    expect(screen.getByText("할일 목록 항목")).toBeInTheDocument();
  });

  it("수정중일때는 수정중일때의 UI가 보여야한다", async () => {
    renderTask();
    await userEvent.click(screen.getByLabelText("수정 버튼"));
    expect(screen.getByDisplayValue("할일 목록 항목")).toBeInTheDocument();
  });

  function renderTask() {
    // 나중에 swr의 renderConfig를 베껴보는건?
    return render(
      <Task
        todo={{
          id: 1,
          title: "할일 목록 항목",
          status: "active",
          start: new Date("2024-03-01"),
          end: new Date("2024-03-15"),
        }}
        onDelete={() => {}}
        onUpdate={() => {}}
      />
    );
  }
});
