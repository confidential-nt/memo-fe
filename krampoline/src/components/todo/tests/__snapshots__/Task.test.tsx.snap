// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Task Test > 스냅샷 테스트 1`] = `
<DocumentFragment>
  <li
    class="flex justify-between items-center w-30 h-12 shrink-0 border-black border-2 rounded-[15px] shadow-standard bg-white my-4"
  >
    <div
      class="flex justify-left items-center"
    >
      <span
        class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorSecondary MuiCheckbox-sizeMedium cursor-pointer css-1sg8fd1-MuiButtonBase-root-MuiCheckbox-root"
      >
        <input
          class="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate="false"
          id="1"
          type="checkbox"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </span>
      <label
        class="flex justify-left items-center"
        for="1"
        style="text-decoration: none;"
      >
        할일 목록 항목
      </label>
    </div>
    <span
      class="flex justify-right items-center"
    >
      <button
        aria-label="수정 버튼"
      >
        <svg
          class="w-6 h-6 shrink-0 mx-1 cursor-pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM5.92 19H5v-.92l9.06-9.06.92.92L5.92 19zM20.71 5.63l-2.34-2.34c-.2-.2-.45-.29-.71-.29s-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83a.996.996 0 000-1.41z"
          />
        </svg>
      </button>
      <button
        aria-label="삭제 버튼"
        class="w-6 h-6 shrink-0 mx-1 cursor-pointer"
      >
        <svg
          aria-hidden="true"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 7l-.867 12.142A2 2 0 0116.138 21H7.862a2 2 0 01-1.995-1.858L5 7m5 4v6m4-6v6m1-10V4a1 1 0 00-1-1h-4a1 1 0 00-1 1v3M4 7h16"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </button>
    </span>
  </li>
</DocumentFragment>
`;
